src/utilstrencodings.h

```diff
@@ -173,4 +173,91 @@ bool ConvertBits(const O& outfn, I it, I end) {
     return true;
 }
 
+/**
+ * Converts the given character to its lowercase equivalent.
+ * This function is locale independent. It only converts uppercase
+ * characters in the standard 7-bit ASCII range.
+ * @param[in] c     the character to convert to lowercase.
+ * @return          the lowercase equivalent of c; or the argument
+ *                  if no conversion is possible.
+ */
+inline unsigned char ToLower(unsigned char c)
+{
+    static const unsigned char mappings[256] = {
```

Comment:
That's a lot of code!

What about `return (c >= 'A' && c <= 'Z' ? (c - 'A') + 'a' : c);`?

Reply:
Thanks @sipa, 

I asked myself the same question and I am happy to change it!

Just to share why I decided to use a mappings array:

**Support for extended ASCII**

Should there be a time where it becomes important to have support for characters in the extended ASCII range then it will be trivial to implement it. It will then *just* be a matter of adjusting entries in mappings array vs extending the implementation with control statements like `if (c == 'Ü') return 'ü';` or alternative logic.

**Performance**

I wanted our locale independent functions to be a good citizen in terms of performance for code that requires it.

I did some naive tests to get an impression what the code would compile to and it seemed that the compiler was able to generate more efficient code in the case `return mappings[c];`

At all optimization levels I would get a pointer to the mappings array where mappings are performed on:
```
lea rax, [rip + __ZZ7ToLowerhE15tolowermappings]
movzx   eax, byte ptr [rdi + rax]
```
Alternatively in the case of `(c >= 'A' && c <= 'Z' ? (c - 'A') + 'a' : c)` the compiler would generate something like this at `O3` optimization:
```
    mov eax, edi 
    add al, -65 
    cmp al, 26
    jae LBB3_2
    add dil, 32
LBB3_2:
    movzx   eax, dil
```
Particularly in the case of iterations I felt the compiler was able to do better optimizations,  for example minimizing iterations by operating in sequences. In the case of the former it could do 4 mapping operations per sequence; while in the latter of at most two mappings per iteration.

On this basis of this I assumed the code would also perform better, but I realize I can be very wrong because I am no expert at this.

I hope this feedback is useful. It it isn't, I will gladly change the code!

Reply:
Thanks @practicalswift, will update the code per @sipa's feedback!

Reply:
I agree with @sipa. If we need to add extended ascii support, we can address that then, and this is not a performance-critical operation that calls for extra code to optimize it. Simple, repetitive code adds visual noise and can hide bugs as well.

Reply:
Thanks @empact, I will update the pull request!

Reply:
Thanks again for the feedback. I have updated the pull request and addressed your comments in 53b3bfa.